test-suite/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -1372,3 +1378,18 @@ object ExportObjSetterNamed_= { // scalastyle:ignore
   @JSExport
   val x = 1
 }
+
+@JSExport
+object ExportFunctionValues {
+  @JSExport
+  def getObj0(): js.Object = new js.Object{val x = "x0"}
+
+  @JSExport
+  def getObj1(): js.Object = new js.Object{val x = "x1"}.asInstanceOf[js.Object]
+
+  @JSExport
+  def getObj2(): js.Object = {
+    val o = new js.Object{val x = "x2"}
+    o
+  }
```

Comment:
Same here: Why do the methods need to be exported? 
Also: Why do you need a separate object in the first place? (What prevents you from just putting the defs in the `if` block?)